ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/datalayer/datalayer.js

```diff
@@ -40,86 +40,131 @@
         return dataLayerContent.component[fieldId];
     }
     
-    const getFieldPanelTitle = (fieldData, dataLayerContent) => {
+    const getFieldPanelTitle = (fieldData, dataLayerContent, formContainerID) => {
         if(fieldData !== undefined){
             if(dataLayerContent.component[fieldData[PARENT_ID]] === undefined) {
-                return dataLayerContent.page[fieldData[PARENT_ID]][DC_TITLE];
+                return containerState[formContainerID].formTitle;
             }
             return dataLayerContent.component[fieldData[PARENT_ID]][DC_TITLE] ;
         } else {
             return '';
         }
     }
 
-    const getEventInfo = (_eventTitle, _formTitle, _fieldId, dataLayerContent) => {
+    const getEventInfo = (_eventTitle, _formTitle, _fieldId, dataLayerContent, formContainerID) => {
         // global variables to be used in abandon event which can be triggered anytime
         const fieldData = getCurrentFieldData(_fieldId, dataLayerContent);
-        formTitle = _formTitle;
-        fieldTitle = fieldData ? fieldData[DC_TITLE] : fieldTitle;
-        fieldType = fieldData ? fieldData[FIELD_TYPE] : fieldType;
-        panelTitle = getFieldPanelTitle(fieldData, dataLayerContent);
+        containerState[formContainerID].formTitle = _formTitle;
+        containerState[formContainerID].fieldTitle = fieldData ? fieldData[DC_TITLE] : containerState[formContainerID].fieldTitle;
+        containerState[formContainerID].fieldType = fieldData ? fieldData[FIELD_TYPE] : containerState[formContainerID].fieldType;
+        containerState[formContainerID].panelTitle = getFieldPanelTitle(fieldData, dataLayerContent, formContainerID);
         return {
-            formTitle,
-            fieldTitle,
-            fieldType,
-            panelTitle,
-            eventName: _eventTitle
+            target: _fieldId,
+            originalTarget: _fieldId,
+            type: _eventTitle,
+            payload: {
+                formTitle: containerState[formContainerID].formTitle,
+                fieldTitle: containerState[formContainerID].fieldTitle,
+                fieldType: containerState[formContainerID].fieldType,
+                panelTitle: containerState[formContainerID].panelTitle,
+                formPath: containerState[formContainerID].formPath,
+            }
         }
     }
 
-    const registerGuideBridgeEvents = (bridge) => {
+    const registerGuideBridgeEvents = (e) => {
+        const formContainerPath = e.detail.getPath();
+        containerState[formContainerPath] = { 
+            submitted: false,
+            formPath: formContainerPath.substring(0, formContainerPath.indexOf("/jcr:content"))
```

Comment:
we should always have the form container path as the form path. It would help in forms in sites use cases too. the path before jcr:content is the path of web page in which form is there.

Reply:
changed the same to send complete comtainerPath.